### DISCUSSION TITLE
Kinesis Advantage 360 Pro: F13-F24 keys do not work on Fn layer

### DISCUSSION THREAD

**Discussion by adamantine-dev** [nickcoutsos/keymap-editor#276] [Q&A] [answered]
Hi Nick! 

Thank you so much for your project! It helps us a lot with configuring ZMK keyboards :)

Recently I have faced an issue when the F13-F24 keys don't work at all.  

My Fn layer is:

<img width="1438" alt="Screenshot 2024-12-29 at 14 07 05" src="https://github.com/user-attachments/assets/e1acbd5d-2ab5-444d-9393-a6121d9c5c2c" />

I am using a MacOS system. 
I tried to press those keys in System Setting, WebStorm IDE, and the following site: 
https://www.toptal.com/developers/keycode

All of those places didn't trigger anything when I tried to hit newly added F13-F24 keys, while everything is good with F1-F12:
(info from https://www.toptal.com/developers/keycode):
- F1 is equal to 112 code;
- F2 is 113;
- ... 
- F12 - 123.

Everything should be according to this table: 
https://www.toptal.com/developers/keycode/table

So, according to it:
- F13 should be 124;
- F14 - 125;
- ...
- F24 - 135.

I could do something wrong, I am still new to ZMK configuration.

Please let me know how to resolve this issue :)

---

PS: is there any way to add F25-F32 keys? 
I have never used them before, but it will be nice to play with them in IDEs HotKeys.

**Comment by nickcoutsos** [comment, marked as answer]
Hey, I think this is a configuration issue that a few Advantage 360 users have run into: https://zmk.dev/docs/config/system#hid (scroll down a bit to the NKRO usages info box).

As for the higher function keys you'd have to add them in ZMK itself first, but I take it they're not actually defined in the HID spec.

**Reply by adamantine-dev** [reply]
Thank you a lot!
Indeed, there is a mention in the main README file of my GitHub repo:

> **N-Key Rollover**
> By default this keyboard has NKRO enabled, however, for compatibility reasons, the higher ranges are not enabled. If you want to use F13-F24 or the INTL1-9 keys with NKRO enabled you can change CONFIG_ZMK_HID_KEYBOARD_EXTENDED_REPORT=n to CONFIG_ZMK_HID_KEYBOARD_EXTENDED_REPORT=y in **_adv360_left_defconfig_**: 
your-repo-link/Adv360-Pro-ZMK/blob/V3.0/config/boards/arm/adv360/adv360_left_defconfig

Once I did it, it worked!
F13-F20 keys are available now.

---

However, the F21-24 keys still don't work. 
Do you know why they don't work? Could it be some MacOS limitations?

One more interesting point: the new F13-F20 keys are working fine on my personal MacBook, but they don't work on my company's MacBook. 
Could it be some additional security settings on the company's laptop?

---

> As for the higher function keys you'd have to add them in ZMK itself first, but I take it they're not actually defined in the HID spec.

Ok, thank you for the information. 
Will read the documentation carefully to understand how to do it.
